aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -19,8 +19,11 @@ static inline Device ensure_has_index(Device device) {
   return impl->getDevice();
 }
 
-static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, bool non_blocking) {
+static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, bool non_blocking, MemoryFormat memory_format=MemoryFormat::Contiguous) {
   auto r = at::empty(self.sizes(), options);
+  if (memory_format == MemoryFormat::ChannelsLast) {
+    r.maybe_as_channels_last();
```

Comment:
Why is it OK to discard the return result here? It looks to me like we should fail if `as_channels_last` failed. (This question is coming from a point of ignorance: I don't actually know what the design spec said should happen in this case.)

Reply:
So, reading into the semantics more, what it looks like will happen here, is if I pass channels last, it will ONLY do something if self was 4-D, and it will silently ignore the memory layout tag if it wasn't. That sounds really bad!

Reply:
putting this implementation in `to_impl` seems premature, given I thought we didn't want it in `to`.  And even if we did want it in to, having its implementation in the function with all the parameters seems questionable because there are a ton of combinations that don't make sense (nchw+ sparse, for example).